gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,360 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.testing.ArbitraryInstances;
+import com.google.gcloud.AuthCredentials;
+import com.google.gcloud.WriteChannel;
+import com.google.gcloud.storage.BlobInfo;
+import com.google.gcloud.storage.BucketInfo;
+import com.google.gcloud.storage.Storage;
+import com.google.gcloud.storage.StorageOptions;
+import com.google.gcloud.storage.testing.RemoteGcsHelper;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.BufferedWriter;
+import java.io.ByteArrayInputStream;
+import java.io.EOFException;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Integration test for gcloud-nio. This test actually talks to GCS (you need an account).
+ * Tests both reading and writing.
+ */
+@RunWith(JUnit4.class)
+public class ITGcsNio {
+
+  //
+  // You *must* set the GOOGLE_APPLICATION_CREDENTIALS environment variable
+  // for this test to work. It must contain the name of a local file that contains
+  // your Service Account JSON Key.
+  //
+  // The instructions for how to get the Service Account JSON Key are
+  // at https://cloud.google.com/storage/docs/authentication?hl=en#service_accounts
+  //
+  // The short version is this: go to cloud.google.com/console,
+  // select your project, search for "API manager", click "Credentials",
+  // click "create credentials/service account key", new service account,
+  // JSON. The contents of the file that's sent to your browsers is your
+  // "Service Account JSON Key".
+  //
+
+  private static final List<String> FILE_CONTENTS = ImmutableList.of(
+      "Tous les êtres humains naissent libres et égaux en dignité et en droits.",
+      "Ils sont doués de raison et de conscience et doivent agir ",
+      "les uns envers les autres dans un esprit de fraternité.");
+
+  private static final Logger log = Logger.getLogger(ITGcsNio.class.getName());
+  private static final String BUCKET = RemoteGcsHelper.generateBucketName();
+  private static final String SML_FILE = "tmp-test-small-file.txt";
+  private static final int SML_SIZE = 100;
+  // it's big, relatively speaking.
+  private static final String BIG_FILE = "tmp-test-big-file.txt";
+  // arbitrary size that's not too round.
+  private static final int BIG_SIZE = 2024 * 1024 - 50;
```

Comment:
Nit, did you mean 2048 * 1024 .... If so, 2 * 1024 * 1024 - 50 would be more clear.